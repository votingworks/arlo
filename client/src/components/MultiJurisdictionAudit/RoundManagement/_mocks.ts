export interface INullResultValues {
  [contestId: string]: {
    [choiceId: string]: null | string | number
  }
}

export const resultsMocks: {
  [key in 'emptyInitial' | 'complete']: INullResultValues
} = {
  emptyInitial: {
    'contest-id-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'contest-id-2': {
      'choice-id-3': null,
      'choice-id-4': null,
    },
  },
  complete: {
    'contest-id-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'contest-id-2': {
      'choice-id-3': 1,
      'choice-id-4': 2,
    },
  },
}

export default resultsMocks
