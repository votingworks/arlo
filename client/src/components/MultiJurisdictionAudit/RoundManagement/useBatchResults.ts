import { useEffect, useState } from 'react'
import { toast } from 'react-toastify'
import { api } from '../../utilities'

export interface IResultValues {
  [batchId: string]: {
    [choiceId: string]: number | string
  }
}

export interface IBatch {
  id: string
  name: string
  numBallots: number
  auditBoard: null | {
    id: string
    name: string
  }
}

const stringifyPossibleNull = (v: string | number | null) => (v ? `${v}` : '')

const reformatResults = (r: IResultValues, numberify = true): IResultValues => {
  return Object.keys(r).reduce(
    (a, batchId) => ({
      ...a,
      [batchId]: Object.keys(r[batchId]).reduce(
        (b, choiceId) => ({
          ...b,
          [choiceId]: numberify
            ? Number(r[batchId][choiceId])
            : stringifyPossibleNull(r[batchId][choiceId]),
        }),
        {}
      ),
    }),
    {}
  )
}

const numberifyResults = (r: IResultValues): IResultValues => reformatResults(r)

const getResults = async (
  electionId: string,
  jurisdictionId: string,
  roundId: string
): Promise<IResultValues | null> => {
  const response = await api<IResultValues>(
    `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches/results`
  )
  if (!response) return null
  return reformatResults(response, false)
}

const getBatches = async (
  electionId: string,
  jurisdictionId: string,
  roundId: string
): Promise<IBatch[] | null> => {
  const response = await api<{ batches: IBatch[] }>(
    `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches`
  )
  if (!response) return null
  return response.batches
}

const useBatchResults = (
  electionId: string,
  jurisdictionId: string,
  roundId: string
): [
  IResultValues | null,
  IBatch[] | null,
  (arg0: IResultValues) => Promise<boolean>
] => {
  const [results, setResults] = useState<IResultValues | null>(null)
  const [batches, setBatches] = useState<IBatch[] | null>(null)

  const updateResults = async (newResults: IResultValues): Promise<boolean> => {
    try {
      await api(
        `/election/${electionId}/jurisdiction/${jurisdictionId}/round/${roundId}/batches/results`,
        {
          method: 'PUT',
          // stringify and numberify the contests (all number values are handled as strings clientside, but are required as numbers serverside)
          body: JSON.stringify(numberifyResults(newResults)),
          headers: {
            'Content-Type': 'application/json',
          },
        }
      )
    } catch (err) /* istanbul ignore next */ {
      // TODO move toasting into api
      toast.error(err.message)
      return false
    }
    setResults(newResults)
    return true
  }

  useEffect(() => {
    ;(async () => {
      const newBatches = await getBatches(electionId, jurisdictionId, roundId)
      const newResults = await getResults(electionId, jurisdictionId, roundId)
      setBatches(newBatches)
      setResults(newResults)
    })()
  }, [electionId, jurisdictionId, roundId])
  return [results, batches, updateResults]
}

export default useBatchResults
