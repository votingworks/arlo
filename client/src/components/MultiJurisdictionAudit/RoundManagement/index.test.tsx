import React from 'react'
import { screen } from '@testing-library/react'
import { Route } from 'react-router-dom'
import { renderWithRouter, withMockFetch } from '../../testUtilities'
import RoundManagement, { IRoundManagementProps } from './index'
import {
  roundMocks,
  batchesMocks,
  batchResultsMocks,
  INullResultValues,
} from './_mocks'
import { dummyBallots } from '../../SingleJurisdictionAudit/_mocks'
import {
  auditSettings,
  auditBoardMocks,
  contestMocks,
} from '../useSetupMenuItems/_mocks'
import { IAuditSettings, IContest } from '../../../types'
import { IBatch } from './useBatchResults'
import { jaApiCalls } from '../_mocks'
import AuthDataProvider from '../../UserContext'

const renderView = (props: IRoundManagementProps) =>
  renderWithRouter(
    <Route
      path="/election/:electionId/jurisdiction/:jurisdictionId"
      render={routeProps => (
        <AuthDataProvider>
          <RoundManagement {...routeProps} {...props} />
        </AuthDataProvider>
      )}
    />,
    {
      route: '/election/1/jurisdiction/jurisdiction-id-1',
    }
  )

const apiCalls = {
  getBallots: {
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/ballots',
    response: dummyBallots,
  },
  getSettings: (response: IAuditSettings) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/settings',
    response,
  }),
  getJAContests: (response: { contests: IContest[] }) => ({
    url: `/api/election/1/jurisdiction/jurisdiction-id-1/contest`,
    response,
  }),
  getBatchResults: (response: INullResultValues) => ({
    url:
      '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/batches/results',
    response,
  }),
  getResults: (response: INullResultValues) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/results',
    response,
  }),
  getBatches: (response: { batches: IBatch[] }) => ({
    url: '/api/election/1/jurisdiction/jurisdiction-id-1/round/round-1/batches',
    response,
  }),
}

describe('RoundManagement', () => {
  it('renders null when still loading', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.blank),
      jaApiCalls.getUser,
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.incomplete,
        auditBoards: [],
        createAuditBoards: jest.fn(),
      })
      expect(container).toMatchSnapshot()
      await screen.findByText('Loading...')
    })
  })

  it('renders audit setup with batch audit', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.batchComparisonAll),
      jaApiCalls.getUser,
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.incomplete,
        auditBoards: [],
        createAuditBoards: jest.fn(),
      })
      await screen.findByText('Round 1 Audit Board Setup')
      expect(container).toMatchSnapshot()
    })
  })

  it('renders audit setup with ballot audit', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.all),
      jaApiCalls.getUser,
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.incomplete,
        auditBoards: [],
        createAuditBoards: jest.fn(),
      })
      await screen.findByText('Round 1 Audit Board Setup')
      expect(container).toMatchSnapshot()
    })
  })

  it('renders complete view', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.all),
      jaApiCalls.getUser,
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.complete,
        auditBoards: auditBoardMocks.signedOff,
        createAuditBoards: jest.fn(),
      })
      await screen.findByText(
        'Congratulations! Your Risk-Limiting Audit is now complete.'
      )
      expect(container).toMatchSnapshot()
    })
  })

  it('renders links & progress with online ballot audit', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.all),
      jaApiCalls.getUser,
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.incomplete,
        auditBoards: auditBoardMocks.unfinished,
        createAuditBoards: jest.fn(),
      })
      await screen.findByText(
        'Download Aggregated Ballot Retrieval List for Round 1'
      )
      expect(container).toMatchSnapshot()
    })
  })

  it('renders links & data entry with offline ballot audit', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.offlineAll),
      jaApiCalls.getUser,
      apiCalls.getJAContests({ contests: contestMocks.oneTargeted }),
      apiCalls.getResults(batchResultsMocks.empty),
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.incomplete,
        auditBoards: auditBoardMocks.unfinished,
        createAuditBoards: jest.fn(),
      })
      await screen.findByText(
        'Download Aggregated Ballot Retrieval List for Round 1'
      )
      expect(container).toMatchSnapshot()
    })
  })

  it('renders links & data entry with batch audit', async () => {
    const expectedCalls = [
      apiCalls.getBallots,
      apiCalls.getSettings(auditSettings.batchComparisonAll),
      jaApiCalls.getUser,
      apiCalls.getJAContests({ contests: contestMocks.oneTargeted }),
      apiCalls.getBatches(batchesMocks.emptyInitial),
      apiCalls.getBatchResults(batchResultsMocks.empty),
    ]
    await withMockFetch(expectedCalls, async () => {
      const { container } = renderView({
        round: roundMocks.incomplete,
        auditBoards: auditBoardMocks.unfinished,
        createAuditBoards: jest.fn(),
      })
      await screen.findByText(
        'Download Aggregated Batch Retrieval List for Round 1'
      )
      expect(container).toMatchSnapshot()
    })
  })
})
