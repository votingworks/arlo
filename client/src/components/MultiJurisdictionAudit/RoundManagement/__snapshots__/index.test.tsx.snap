// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoundManagement renders audit setup with ballot audit 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-ckVGcZ eyWwvb"
  >
    <h3
      class="bp3-heading"
    >
      Round 
      1
       Audit Board Setup
    </h3>
    <p
      class="sc-eNQAEJ iwowog"
    >
      27
       ballots to audit in Round 
      1
    </p>
    <div
      class="sc-dnqmqq jnbCEU"
    >
      <h4
        class="bp3-heading"
      >
        Number of Audit Boards
      </h4>
      <p>
        Select the appropriate number of audit boards based upon the personnel available and the number of ballots assigned to your jurisdiction for this round of the audit. You will have the opportunity to adjust the number of audit boards before the next round of the audit, if another round is required.
      </p>
      <div
        class="sc-bZQynM ibvdum"
      >
        <div
          class="bp3-html-select sc-iwsKbI iSMuWx sc-EHOje lcjkJq"
        >
          <select
            data-testid="numAuditBoards"
            field="[object Object]"
            form="[object Object]"
            id="numAuditBoards"
          >
            <option>
              1
            </option>
            <option>
              2
            </option>
            <option>
              3
            </option>
            <option>
              4
            </option>
            <option>
              5
            </option>
            <option>
              6
            </option>
            <option>
              7
            </option>
            <option>
              8
            </option>
            <option>
              9
            </option>
            <option>
              10
            </option>
            <option>
              11
            </option>
            <option>
              12
            </option>
            <option>
              13
            </option>
            <option>
              14
            </option>
            <option>
              15
            </option>
          </select>
          <span
            class="bp3-icon bp3-icon-double-caret-vertical"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <button
        class="bp3-button bp3-intent-primary sc-ifAKCX zdCNs"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`RoundManagement renders audit setup with batch audit 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-ckVGcZ eyWwvb"
  >
    <h3
      class="bp3-heading"
    >
      Round 
      1
       Audit Board Setup
    </h3>
    <p
      class="sc-eNQAEJ iwowog"
    >
      27
       ballots to audit in Round 
      1
    </p>
    <div
      class="sc-dnqmqq jnbCEU"
    >
      <h4
        class="bp3-heading"
      >
        Number of Audit Boards
      </h4>
      <p>
        Select the appropriate number of audit boards based upon the personnel available and the number of ballots assigned to your jurisdiction for this round of the audit. You will have the opportunity to adjust the number of audit boards before the next round of the audit, if another round is required.
      </p>
      <div
        class="sc-bZQynM ibvdum"
      >
        <div
          class="bp3-html-select sc-iwsKbI iSMuWx sc-EHOje lcjkJq"
        >
          <select
            data-testid="numAuditBoards"
            field="[object Object]"
            form="[object Object]"
            id="numAuditBoards"
          >
            <option>
              1
            </option>
            <option>
              2
            </option>
            <option>
              3
            </option>
            <option>
              4
            </option>
            <option>
              5
            </option>
            <option>
              6
            </option>
            <option>
              7
            </option>
            <option>
              8
            </option>
            <option>
              9
            </option>
            <option>
              10
            </option>
            <option>
              11
            </option>
            <option>
              12
            </option>
            <option>
              13
            </option>
            <option>
              14
            </option>
            <option>
              15
            </option>
          </select>
          <span
            class="bp3-icon bp3-icon-double-caret-vertical"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <button
        class="bp3-button bp3-intent-primary sc-ifAKCX zdCNs"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`RoundManagement renders complete view 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-ckVGcZ eyWwvb"
  >
    <h2
      class="bp3-heading"
    >
      Congratulations! Your Risk-Limiting Audit is now complete.
    </h2>
  </div>
</div>
`;

exports[`RoundManagement renders links & data entry with batch audit 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-ckVGcZ eyWwvb"
  >
    <h3
      class="bp3-heading"
    >
      Round 
      1
       Data Entry
    </h3>
    <div
      class="sc-jKJlTe lcqeCu"
    >
      <p
        class="sc-eNQAEJ iwowog"
      >
        27
         ballots to audit in Round 
        1
      </p>
      <div
        class="bp3-button-group bp3-vertical bp3-align-left"
      >
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-th"
            icon="th"
          >
            <svg
              data-icon="th"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                th
              </desc>
              <path
                d="M15 1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V2c0-.5-.4-1-1-1zM6 13H2v-2h4v2zm0-3H2V8h4v2zm0-3H2V5h4v2zm8 6H7v-2h7v2zm0-3H7V8h7v2zm0-3H7V5h7v2z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Aggregated
             
            Batch
             
            Retrieval List
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-document"
            icon="document"
          >
            <svg
              data-icon="document"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                document
              </desc>
              <path
                d="M9 0H3c-.55 0-1 .45-1 1v14c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5L9 0zm3 14H4V2h4v4h4v8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Placeholder Sheets
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-label"
            icon="label"
          >
            <svg
              data-icon="label"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                label
              </desc>
              <path
                d="M11 2H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h14c.55 0 1-.45 1-1V7l-5-5zm3 10H2V4h8v2H3v1h7v1h4v4zm-3-5V4l3 3h-3zm-8 3h10V9H3v1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Ballot Labels
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-jKJlTe lcqeCu"
    >
      <form>
        <p>
          When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
        </p>
        <div
          class="bp3-card bp3-elevation-2 sc-jTzLTM kZeJjn"
        >
          <h5
            class="bp3-heading"
          >
            Batch: Batch One, Contest: Contest 1
          </h5>
          <label
            class="sc-dxgOiQ hkjLkh"
            for="results[batch-1][choice-id-1]"
          >
            Votes for 
            Choice One
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[batch-1][choice-id-1]"
                  name="results[batch-1][choice-id-1]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            class="sc-dxgOiQ hkjLkh"
            for="results[batch-1][choice-id-2]"
          >
            Votes for 
            Choice Two
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[batch-1][choice-id-2]"
                  name="results[batch-1][choice-id-2]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
        </div>
        <div
          class="bp3-card bp3-elevation-2 sc-jTzLTM kZeJjn"
        >
          <h5
            class="bp3-heading"
          >
            Batch: Batch Two, Contest: Contest 1
          </h5>
          <label
            class="sc-dxgOiQ hkjLkh"
            for="results[batch-2][choice-id-1]"
          >
            Votes for 
            Choice One
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[batch-2][choice-id-1]"
                  name="results[batch-2][choice-id-1]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            class="sc-dxgOiQ hkjLkh"
            for="results[batch-2][choice-id-2]"
          >
            Votes for 
            Choice Two
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[batch-2][choice-id-2]"
                  name="results[batch-2][choice-id-2]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
        </div>
        <div
          class="bp3-card bp3-elevation-2 sc-jTzLTM kZeJjn"
        >
          <h5
            class="bp3-heading"
          >
            Batch: Batch Three, Contest: Contest 1
          </h5>
          <label
            class="sc-dxgOiQ hkjLkh"
            for="results[batch-3][choice-id-1]"
          >
            Votes for 
            Choice One
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[batch-3][choice-id-1]"
                  name="results[batch-3][choice-id-1]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            class="sc-dxgOiQ hkjLkh"
            for="results[batch-3][choice-id-2]"
          >
            Votes for 
            Choice Two
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[batch-3][choice-id-2]"
                  name="results[batch-3][choice-id-2]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-kpOJdX glAiLs sc-ifAKCX zdCNs"
          type="submit"
        >
          <span
            class="bp3-button-text"
          >
            Submit Data for Round 1
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`RoundManagement renders links & data entry with offline ballot audit 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-ckVGcZ eyWwvb"
  >
    <h3
      class="bp3-heading"
    >
      Round 
      1
       Data Entry
    </h3>
    <div
      class="sc-jKJlTe lcqeCu"
    >
      <p
        class="sc-eNQAEJ iwowog"
      >
        27
         ballots to audit in Round 
        1
      </p>
      <div
        class="bp3-button-group bp3-vertical bp3-align-left"
      >
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-th"
            icon="th"
          >
            <svg
              data-icon="th"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                th
              </desc>
              <path
                d="M15 1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V2c0-.5-.4-1-1-1zM6 13H2v-2h4v2zm0-3H2V8h4v2zm0-3H2V5h4v2zm8 6H7v-2h7v2zm0-3H7V8h7v2zm0-3H7V5h7v2z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Aggregated
             
            Ballot
             
            Retrieval List
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-document"
            icon="document"
          >
            <svg
              data-icon="document"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                document
              </desc>
              <path
                d="M9 0H3c-.55 0-1 .45-1 1v14c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5L9 0zm3 14H4V2h4v4h4v8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Placeholder Sheets
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-label"
            icon="label"
          >
            <svg
              data-icon="label"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                label
              </desc>
              <path
                d="M11 2H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h14c.55 0 1-.45 1-1V7l-5-5zm3 10H2V4h8v2H3v1h7v1h4v4zm-3-5V4l3 3h-3zm-8 3h10V9H3v1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Ballot Labels
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-jKJlTe lcqeCu"
    >
      <form>
        <p>
          When you have examined all the ballots assigned to you, enter the number of votes recorded for each candidate/choice from the audited ballots.
        </p>
        <div
          class="bp3-card bp3-elevation-2 sc-jTzLTM kZeJjn"
        >
          <h5
            class="bp3-heading"
          >
            Contest 1
          </h5>
          <label
            class="sc-kGXeez htRkSS"
            for="results[contest-id-1][choice-id-1]"
          >
            Votes for 
            Choice One
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[contest-id-1][choice-id-1]"
                  name="results[contest-id-1][choice-id-1]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            class="sc-kGXeez htRkSS"
            for="results[contest-id-1][choice-id-2]"
          >
            Votes for 
            Choice Two
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[contest-id-1][choice-id-2]"
                  name="results[contest-id-1][choice-id-2]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
        </div>
        <div
          class="bp3-card bp3-elevation-2 sc-jTzLTM kZeJjn"
        >
          <h5
            class="bp3-heading"
          >
            Contest 2
          </h5>
          <label
            class="sc-kGXeez htRkSS"
            for="results[contest-id-2][choice-id-3]"
          >
            Votes for 
            Choice Three
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[contest-id-2][choice-id-3]"
                  name="results[contest-id-2][choice-id-3]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
          <label
            class="sc-kGXeez htRkSS"
            for="results[contest-id-2][choice-id-4]"
          >
            Votes for 
            Choice Four
            :
            <div
              class="sc-fjdhpX hpEPln"
            >
              <div
                class="bp3-input-group sc-jzJRlG lpfNbV"
              >
                <input
                  class="bp3-input"
                  id="results[contest-id-2][choice-id-4]"
                  name="results[contest-id-2][choice-id-4]"
                  style="padding-right: 10px;"
                  value=""
                />
              </div>
            </div>
          </label>
        </div>
        <button
          class="bp3-button bp3-intent-primary sc-kgoBCf oMZJm sc-ifAKCX zdCNs"
          type="submit"
        >
          <span
            class="bp3-button-text"
          >
            Submit Data for Round 1
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`RoundManagement renders links & progress with online ballot audit 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-ckVGcZ eyWwvb"
  >
    <h3
      class="bp3-heading"
    >
      Round 
      1
       Data Entry
    </h3>
    <div
      class="sc-jKJlTe lcqeCu"
    >
      <p
        class="sc-eNQAEJ iwowog"
      >
        27
         ballots to audit in Round 
        1
      </p>
      <div
        class="bp3-button-group bp3-vertical bp3-align-left"
      >
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-th"
            icon="th"
          >
            <svg
              data-icon="th"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                th
              </desc>
              <path
                d="M15 1H1c-.6 0-1 .5-1 1v12c0 .6.4 1 1 1h14c.6 0 1-.4 1-1V2c0-.5-.4-1-1-1zM6 13H2v-2h4v2zm0-3H2V8h4v2zm0-3H2V5h4v2zm8 6H7v-2h7v2zm0-3H7V8h7v2zm0-3H7V5h7v2z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Aggregated
             
            Ballot
             
            Retrieval List
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-document"
            icon="document"
          >
            <svg
              data-icon="document"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                document
              </desc>
              <path
                d="M9 0H3c-.55 0-1 .45-1 1v14c0 .55.45 1 1 1h10c.55 0 1-.45 1-1V5L9 0zm3 14H4V2h4v4h4v8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Placeholder Sheets
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-label"
            icon="label"
          >
            <svg
              data-icon="label"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                label
              </desc>
              <path
                d="M11 2H1c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h14c.55 0 1-.45 1-1V7l-5-5zm3 10H2V4h8v2H3v1h7v1h4v4zm-3-5V4l3 3h-3zm-8 3h10V9H3v1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Ballot Labels
          </span>
        </button>
        <button
          class="bp3-button"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-key"
            icon="key"
          >
            <svg
              data-icon="key"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                key
              </desc>
              <path
                d="M11 0C8.24 0 6 2.24 6 5c0 1.02.31 1.96.83 2.75L.29 14.29a1.003 1.003 0 001.42 1.42L3 14.41l1.29 1.29c.18.19.43.3.71.3s.53-.11.71-.29l2-2c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71L6.41 11l1.83-1.83c.8.52 1.74.83 2.76.83 2.76 0 5-2.24 5-5s-2.24-5-5-5zm0 8c-.23 0-.45-.03-.66-.08-.01 0-.02-.01-.03-.01-.21-.05-.41-.12-.6-.21a3.014 3.014 0 01-1.62-2c0-.01-.01-.02-.01-.03C8.03 5.45 8 5.23 8 5c0-1.66 1.34-3 3-3s3 1.34 3 3-1.34 3-3 3z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            Download Audit Board Credentials
          </span>
        </button>
        <div
          class="sc-VigVT iyEYjY"
          id="qr-root"
        >
          <span
            id="qr-happy-rebel-base"
          >
            <canvas
              height="200"
              style="height: 200px; width: 200px;"
              width="200"
            />
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-jKJlTe lcqeCu"
    >
      <div
        class="sc-gZMcBi hUnsTv"
      >
        <h4
          class="bp3-heading"
        >
          Audit Board Progress
        </h4>
        <span>
          0
           of 
          30
           ballots audited
           
        </span>
        <div
          class="bp3-progress-bar bp3-intent-primary"
        >
          <div
            class="bp3-progress-meter"
            style="width: 0%;"
          />
        </div>
      </div>
      <div
        class="sc-gqjmRU fWgawt"
      >
        <span>
          Audit Board #01: 0 of 30 ballots audited 
        </span>
        <div
          class="bp3-progress-bar bp3-intent-primary"
        >
          <div
            class="bp3-progress-meter"
            style="width: 0%;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
